stdlib/Base64/src/encode.jl

```diff
@@ -2,17 +2,36 @@
 
 # Generate encode table.
 const BASE64_ENCODE = [UInt8(x) for x in append!(['A':'Z'; 'a':'z'; '0':'9'], ['+', '/'])]
-encode(x::UInt8) = @inbounds return BASE64_ENCODE[(x & 0x3f) + 1]
-encodepadding()  = UInt8('=')
+const BASE64URL_ENCODE = [UInt8(x) for x in append!(['A':'Z'; 'a':'z'; '0':'9'], ['-', '_'])]
+const TABLE_ENCODE = vcat(BASE64_ENCODE, BASE64URL_ENCODE)
 
+const ENCODEPADDING  = UInt8('=')
+
+"""
+    Base64Format
+
+Enum to declare encoding and decoding format.
+
++ `BASE64`: Stands for Base64 format.
++ `BASE64URL`: Stands for Base64URL format.
 """
-    Base64EncodePipe(ostream)
+@enum Base64Format::UInt8 BASE64=0 BASE64URL=1
```

Comment:
Why not have the values be `0` and `256` so that you don't need to do the `decode << 8` shift when they are used?

Reply:
Oh, I see — the shift for the decode and encode tables is different.

But in that case, it seems clearer to use 2d arrays rather than concatenated 1d arrays.   i.e. `TABLE[char, decode]` rather than `TABLE[char + (decode << x)]`?   And then make the `@enum` values `1` or `2`.

Reply:
this was also done in [3142dd7](https://github.com/JuliaLang/julia/pull/49849/commits/3142dd70e1c9af95b6a83d3c1798433680948b04).